fonts/generate-smufl-code.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python3
+
+import os
+from lxml import etree
+
+# Define file output paths
+smufl_h_path = '../include/vrv/smufl.h'
+smufl_cpp_path = '../src/smufl.cpp'
+
+######################
+#  Helper Functions  #
+######################
+
+# Open file in same directory as script and retrieve its content
+def get_file_content(filepath):
+   __location__ = os.path.realpath(os.path.dirname(__file__))
+   file = open(os.path.join(__location__, filepath), 'r')
+   content = file.read()
+   file.close()
+   return content
+
+# Write content to file with path relative to the script directory
+def write_file_content(filepath, content):
+   __location__ = os.path.realpath(os.path.dirname(__file__))
+   file = open(os.path.join(__location__, filepath), 'w')
+   file.write(content)
+   file.close()
+
+# Retrieve all elements with given tag name from xml
+def get_elements(xml, tag):
+   root = etree.fromstring(bytes(xml, encoding='utf-8'))
+   return root.findall('.//' + tag)  # XPath, recursive
+
+#########################
+#  Parse supported.xsl  #
+#########################
+
+# Retrieve dictionary mapping glyph code to smufl name
+def get_supported_glyph_codes():
+   xml = get_file_content('supported.xsl')
+   glyphs = get_elements(xml, 'glyph')
+   dict = {}
+   for glyph in glyphs:
+      dict[glyph.attrib["glyph-code"]] = glyph.attrib["smufl-name"]
+   return dict
+
+##################
+#  Main program  #
```

Comment:
Surely this can be collapsed into the previous script, and configured with some sort of mode flag? All of the 'helper functions' are exactly the same.

Reply:
Also, this should be in a `if __name__ == "__main__"` block

Reply:
I would prefer to keep the scripts separate as they have different purpose.